binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -3,11 +3,26 @@
 
 import filecmp
 import os
+import re
 import sys
 from argparse import ArgumentParser
 
-from ..reporter import exception_reporting
 from ._parser import add_version_argument_to
+from ..reporter import exception_reporting
+
+keywords_pattern = re.compile('KEYWORDS="(?P<keywords>.*?)"')
+
+
+def _get_relevant_keywords_set_for(ebuild_content, expected_keywords):
+    try:
+        ebuild_keywords = keywords_pattern.search(ebuild_content).group('keywords')
+    except AttributeError:
+        # if the KEYWORDS variable is not found in the ebuild, we will assume the ebuild needs to be listed
+        ebuild_keywords = expected_keywords
+    expected_keywords_set = set(expected_keywords.split(" "))
+    ebuild_keywords_set = set(ebuild_keywords.split(" "))
+
+    return expected_keywords_set & ebuild_keywords_set
```

Comment:
I wonder if that is the "best math" already. If the user asks for "~amd64" they probably want to get stable packages too without asking for "amd64 ~amd64" explicitly, similar to how `ACCEPT_KEYWORD` works in portage. Is something like that happening somewhere?

Reply:
No, the current code doesn't take int account that ~amd64 could implicitly include amd64 too. I haven't been using testing keywords a lot, so I didn't even know it worked like that. If you want, I can replicate this behavior. If portage is doing it, it is probably what the user expects so it makes sense.

Reply:
Okay cool.  If I had to implement it, I'd do something like this, just an idea:
```
In [14]: keywords_flat = '~amd64 ~x86'  # example

In [15]: keywords = {k for k in keywords_flat.split(' ') if k}  # be robust towards multiple spaces

In [16]: keywords |= {k[1:] for k in keywords if k.startswith('~')}  # add stable to unstable

In [17]: keywords
Out[17]: {'amd64', 'x86', '~amd64', '~x86'}
```